web/src/compose_recipient.js

```diff
@@ -3,25 +3,35 @@
 import $ from "jquery";
 import _ from "lodash";
 
+import render_inline_decorated_stream_name from "../templates/inline_decorated_stream_name.hbs";
+
 import * as compose_banner from "./compose_banner";
 import * as compose_fade from "./compose_fade";
 import * as compose_pm_pill from "./compose_pm_pill";
 import * as compose_state from "./compose_state";
 import * as compose_ui from "./compose_ui";
 import * as compose_validate from "./compose_validate";
-import {DropdownListWidget} from "./dropdown_list_widget";
 import {$t} from "./i18n";
 import * as narrow_state from "./narrow_state";
-import {page_params} from "./page_params";
-import * as settings_config from "./settings_config";
 import * as stream_bar from "./stream_bar";
 import * as stream_data from "./stream_data";
+import * as stream_dropdown_list from "./stream_dropdown_list";
 import * as ui_util from "./ui_util";
 import * as util from "./util";
 
-export let compose_recipient_widget;
+export let selected_stream_name;
```

Comment:
We probably want to store the selected stream by it's ID, that's robust to stream name editing automatically; would also solve the namespace problem, as the value can be either `6` or `"direct"`.

Reply:
Done